packages/devtools_app/lib/src/flutter/device_dialog.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:pedantic/pedantic.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../connected_app.dart';
+import '../globals.dart';
+import '../info/info_controller.dart';
+import '../table_data.dart';
+import '../utils.dart';
+import '../version.dart';
+import 'auto_dispose_mixin.dart';
+import 'common_widgets.dart';
+import 'table.dart';
+import 'theme.dart';
+
+class DeviceDialog extends StatelessWidget {
+  const DeviceDialog({
+    @required this.connectedApp,
+    @required this.flutterVersion,
+  });
+
+  final ConnectedApp connectedApp;
+  final FlutterVersion flutterVersion;
+
+  @override
+  Widget build(BuildContext context) {
+    const boldText = TextStyle(fontWeight: FontWeight.bold);
+    final theme = Theme.of(context);
+
+    final vm = serviceManager.vm;
+
+    var version = vm.version;
+    if (version.contains(' ')) {
+      version = version.substring(0, version.indexOf(' '));
+    }
+
+    final versions = {
+      'Dart Version': version,
+      'CPU / OS':
+          '${vm.targetCPU}-${vm.architectureBits} / ${vm.operatingSystem}',
+    };
+
+    if (flutterVersion != null) {
+      versions['Flutter Version'] =
+          '${flutterVersion.version} / ${flutterVersion.channel}';
+      versions['Framework / Engine'] = '${flutterVersion.frameworkRevision} / '
+          '${flutterVersion.engineRevision}';
+    }
+
+    return AlertDialog(
+      actions: [
+        if (connectedApp.isRunningOnDartVM) _ViewVMFlagsButton(),
+        DialogCloseButton(),
+      ],
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          ...headerInColumn(theme.textTheme, 'Device Info'),
+          for (var name in versions.keys)
+            Padding(
+              padding: const EdgeInsets.only(bottom: denseRowSpacing),
+              child: Row(
+                children: [
+                  Text('$name: ', style: boldText),
+                  Text(versions[name], style: theme.subtleTextStyle),
+                ],
+              ),
+            ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ViewVMFlagsButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      onPressed: () {
+        Navigator.of(context, rootNavigator: true).pop('dialog');
+
+        unawaited(showDialog(
+          context: context,
+          builder: (context) => VMFlagsDialog(),
+        ));
+      },
+      child: Text('View VM Flags...'.toUpperCase()),
```

Comment:
Why the ellipsis?

Reply:
To indicate that it opens a separate dialog (instead of just doing something like performing an action).